static/js/components/dashboard/courses/StatusMessages.js

```diff
@@ -133,8 +134,41 @@ export const calculateMessages = (props: CalculateMessagesProps) => {
     if (hasPaidForAnyCourseRun(course)) {
       // exam is required, user has not yet passed it
       if (exams && ! passedExam) {
+        let messageBox = {};
+        if (failedExam) {
```

Comment:
I missed this on my first review pass, but it seems like if you ever failed an exam, you'll get into this code (even if you subsequently pass the exam), which I don't think we want. I think the check on line 136 (`!passedExam`) is all we need here.